### PR TITLE
[incubator/elastic-stack] Use stable version of fluentd-elasticsearch for elastic-stack

### PR DESCRIPTION
**Pull request by rrader**
@rendhalver @jar361 @christian-roggia 

#### What this PR does / why we need it:

stable/fluentd-elasticsearch is better than incubator/fluentd because it creates DaemonSet and works out of box without much additional configuration.

#### Special notes for your reviewer:

I tested this chart on our cluster.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: incubator/elastic-stack/requirements.lock [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-  version: 0.13.1
+  version: 0.14.6
 - name: logstash
   repository: https://kubernetes-charts-incubator.storage.googleapis.com/
-  version: 0.6.3
-- name: fluentd
-  repository: https://kubernetes-charts-incubator.storage.googleapis.com/
-  version: 0.1.4
+  version: 0.9.3
+- name: fluentd-elasticsearch
```
Comment (rendhalver) [inline incubator/elastic-stack/requirements.lock:11]: Why the change?
fluentd and fluentd-elasticsearch are two different things.
fluentd is for ingesting logs from external sources and fluentd-elasticsearch is for ingesting container logs from nodes.

**Review by rendhalver — CHANGES_REQUESTED** [review]

File: incubator/elastic-stack/requirements.lock (hunk 1 above)
Comment (rrader) [reply incubator/elastic-stack/requirements.lock:11]: Why different? As I understood, fluentd deploys the single instance of fluentd, and fluentd-elasticsearch deploys the DaemonSet with fluentd on each node. Also, it has configuration to gather logs from all containers on node.

Single fluentd itself is not helpful while deploying the elastic-stack, IMHO. What I would want from this chart is the working EFK for all my containers.

Comment (desaintmartin) [reply incubator/elastic-stack/requirements.lock:11]: So maybe we need options to both, actually.

Comment (rendhalver) [reply incubator/elastic-stack/requirements.lock:11]: Look at the two charts.
We use the fluentd Chart for ingesting external logs into Elastic-Stack in one cluster and we also use fluentd-elasticstack to ingest container logs into elastic-stack into a different cluster.

Comment (desaintmartin) [reply incubator/elastic-stack/requirements.lock:11]: my point is that, in order to cover most use cases for elastic-stack, we would need to have fluentd AND fluentd-elasticsearch subcharts.

Actually, I already do this with custom charts. Doing it from upstream would be nice.

Comment (rrader) [reply incubator/elastic-stack/requirements.lock:11]: Yeah, you right. I'll change PR to include both cases

Comment (rendhalver) [reply incubator/elastic-stack/requirements.lock:11]: Sounds good to me.

**Comment by desaintmartin** [comment]
Should we finish this or close it?
